packages/util/src/unique-name-generator.ts

```diff
@@ -0,0 +1,214 @@
+/**
```

Comment:
Since the file was moved, what was the small error you fixed?

Reply:
Sorry, yes you cannot see this anymore... -.-

To generate the final name, we internally get the name without index using a regex. This regex had `'_'` hardcoded as a separator. As you can set different separators, the following test would fail for `'-'`:
```ts
it('should add -2 if -1 is already used', () => {
    const generator = new UniqueNameGenerator('-', ['MyClass-1']);
    expect(generator.generateUniqueName('MyClass-1')).toBe('MyClass-2');
});
```
The result was `'MyClass-1_1'` before.

Now that I think of it, this technically is a breaking change. `-` is used for directory names and npm package names, but for existing projects those respect the `service-mapping.json`. We might want to mention it in the compatibility notes...

Reply:
yes